cmake/externals/dlib.cmake

```diff
@@ -1,6 +1,8 @@
 FetchContent_Declare(dlib
     GIT_REPOSITORY https://github.com/davisking/dlib.git
-    GIT_TAG        v19.22
+    # there is non an official tag which supports STFT,
+    # choose a relatively stable commit id for that.
+    GIT_TAG        a12824d42584e292ecb3bad05c4b32c2015a7b89
```

Comment:
This might be problematic when we have to switch to using a URL/URL_HASH based setup as we need a zip from a release.

Reply:
A not-well-known feature is github can also produce zip for any commit id, which is the way how ort integrate onnx and ort-extensions now.